packages/proxy/src/utils.ts

```diff
@@ -26,20 +26,48 @@ async function getFileHash(filePath: string) {
   return createHash("md5").update(fileBuffer).digest("hex");
 }
 
-export async function createFileWatcher(filePath: string, onChange: () => void) {
-  try {
-    const configWatcher = fs.watch(filePath);
-    let previousFileHash = getFileHash(filePath);
-
-    for await (const event of configWatcher) {
-      if (event.eventType !== "change") continue;
-      const newFileHash = getFileHash(filePath);
-      if (previousFileHash !== newFileHash) {
-        onChange();
-        previousFileHash = newFileHash;
+function withResolvers() {
```

Comment:
Nit: `withResolvers` here is "stolen" from `Promise.prototype.withResolvers` but as a freely named function `withResolvers` is a weird silly name. `createDeferred` is probably more appropriate. I don't care enough to reject this but a lovely cleanup task